### PR TITLE
#149 Refactored integration-tests

### LINKED ISSUE #148: Use virtual-schema-common-documents 3.0.0

### LINKED ISSUE #149: Refactor integration tests
* [x] use hamcrest-result-set-matchers

### PR DESCRIPTION
**Pull request by jakobbraun**
Closes #149
Closes #148

### CODE REVIEW

**Review by morazow — CHANGES_REQUESTED** [review]

File: src/test/java/com/exasol/dynamodb/DynamodbContainer.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+    public DynamodbContainer() {
+        super(DYNAMODB_IMAGE);
+        this.withExposedPorts(DYNAMODB_PORT).withCommand("-jar DynamoDBLocal.jar -sharedDb -dbPath .");
+    }
+
+    /**
+     * Get the port under which DynamoDB is reachable at the host
+     * 
+     * @return port dynamodb is maped to
```
[the remark this answers is not in the document]
Comment (jakobbraun) [reply src/test/java/com/exasol/dynamodb/DynamodbContainer.java:17]:
```suggestion
     * @return port DynamoDB is mapped to
```

File: src/main/java/com/exasol/dynamodb/DynamodbConnectionFactory.java [hunk 2]
```diff
@@ -38,20 +39,32 @@ public static String buildPassWithTokenSeparator(final String key, final Optiona
...
         final AwsCredentials awsCredentials = getAwsCredentials(user, key, sessionToken);
         clientBuilder.credentialsProvider(StaticCredentialsProvider.create(awsCredentials));
         return clientBuilder.build();
     }
 
+    private URI parseDynamodbUri(final String uri) {
+        try {
+            return new URI(uri);
+        } catch (final URISyntaxException exception) {
+            throw new IllegalArgumentException(ExaError.messageBuilder("E-VSDY-1")
```
[the remark this answers is not in the document]
Comment (jakobbraun) [reply src/main/java/com/exasol/dynamodb/DynamodbConnectionFactory.java:60]: no. dynamodb virtual schema has vsdy. vsd is virtual-schema-common-document

File: pom.xml [hunk 3]
```diff
@@ -325,6 +351,12 @@
                         </goals>
                     </execution>
                 </executions>
+                <configuration>
+                    <excludeVulnerabilityIds>
+                        <!-- Ignores CVE-2020-15250, because we use Java 11 and junit 4.13.1 which contains a fix: https://ossindex.sonatype.org/vuln/7ea56ad4-8a8b-4e51-8ed9-5aad83d8efb1 -->
+                        <exclude>7ea56ad4-8a8b-4e51-8ed9-5aad83d8efb1</exclude>
+                    </excludeVulnerabilityIds>
+                </configuration>
```
Comment (morazow) [inline pom.xml:359]: Now, this is fixed, I think you can also remove it here.

**Review by morazow — APPROVED** [review]
Remove the fixed oss-index vulnerability, then good to merge!

**Pull request merged**